geode-core/src/integrationTest/java/org/apache/geode/cache/query/functional/IndexOperatorJUnitTest.java

```diff
@@ -145,7 +145,7 @@ public void testWithNULL() throws Exception {
     map.put("0", new Integer(11));
     map.put("1", new Integer(12));
     Object result = runQuery(map, null);
-    if (result != null)
+    if (result != QueryService.UNDEFINED)
```

Comment:
I realize that you're editing code that isn't great but `fail();` is a terrible anti-pattern in testing.

Try this:
```
assertThat(result).isNotEqualTo(QueryService.UNDEFINED);
```